### PR TITLE
fix(disable-row): allow disabling through side effects without mutating rows

### PR DESCRIPTION
**Pull request by chintankavathia**
**What kind of change does this PR introduce?** (check one with "x")

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by chintankavathia** [comment]
@fh1ch Could you please review this MR?

File: projects/ngx-datatable/src/lib/components/body/body-row-wrapper.component.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export class DataTableRowWrapperComponent implements DoCheck, OnInit {
   @Input() detailRowHeight: any;
   @Input() row: any;
   @Input() groupedRows: any;
-  @Input() disableCheck: (row: any) => boolean;
+  @Input() disableCheck: (row: any) => boolean | BehaviorSubject<boolean>;
```
Comment (fh1ch) [inline projects/ngx-datatable/src/lib/components/body/body-row-wrapper.component.ts:55]: I'm questioning the solution approach a bit: instead of fixing the actual problem, this MR *introduces* an alternative approach to set a row as `disabled`. This means that:

- The commit-message is wrong and this is rather a `feat:`.
- The API documentation in https://github.com/siemens/ngx-datatable/blob/master/docs/api/table/inputs.md shall be updated.
- The new feature should be covered at least by a minimal unit-test within https://github.com/siemens/ngx-datatable/blob/master/projects/ngx-datatable/src/lib/components/datatable.component.spec.ts.

However, I see the bug in another area: the `disableCheck` for the `DataTableRowWrapperComponent` component is executed **within** a `if (this.rowDiffer.diff(this.row)) {`, meaning only if the data changes. Moving it out of said `if` would run it during **every** `ngDoCheck()`, which already resolves the issue **without** introducing something new and on top.

@timowolf @kfenner WDYT? Do you agree with this analysis or shall we introduce an alternative API? However, this would mean that the original bug still persists when using the API the "old" way.

**Review by fh1ch — CHANGES_REQUESTED** [review]
@chintankavathia thanks a lot for the MR, we highly appreciate this 🙇 

One comment from my end regarding the chosen solution approach. Code-wise, this one looks fine, but would miss some other aspects. Back to you 🏓

File: projects/ngx-datatable/src/lib/components/body/body-row-wrapper.component.ts (hunk 1 above)
Comment (kfenner) [reply projects/ngx-datatable/src/lib/components/body/body-row-wrapper.component.ts:55]: @fh1ch Let's fix the bug instead of a new API.

Comment (chintankavathia) [reply projects/ngx-datatable/src/lib/components/body/body-row-wrapper.component.ts:55]: @fh1ch Thanks for the feedback. As you mentioned if we need to just fix the bug then we can move `disableCheck` outside of `if (this.rowDiffer.diff(this.row)) {` of `ngDoCheck()` but here I am worried about the performance aspect as `ngDoCheck` triggers on every change detections and will call `disableCheck` function for each row with every `ngDoCheck` call.

Comment (chintankavathia) [reply projects/ngx-datatable/src/lib/components/body/body-row-wrapper.component.ts]: Fixed by moving out `disableCheck` to call upon every `ngDoCheck`

**Review by fh1ch — APPROVED** [review]
@chintankavathia thanks a ton for the rework, much appreciated 🙇 

I gave it some spins locally and can confirm that things now work as expected. I also did a bit of performance testing without any major negative impact I could identify.

LGTM 👍
